apps/volunteer/choose_roles.py

```diff
@@ -108,3 +120,62 @@ def role(role_id):
         role=role,
         current_volunteer=current_volunteer,
     )
+
+
+@decorator
+def role_admin_required(f, *args, **kwargs):
+    if current_user.is_authenticated:
+        if int(args[0]) in [
+            ra.role_id for ra in current_user.volunteer_admin_roles
+        ] or current_user.has_permission("volunteer:admin"):
+            return f(*args, **kwargs)
+        abort(404)
+    return app.login_manager.unauthorized()
+
+
+@volunteer.route("role/<role_id>/admin")
```

Comment:
I believe these parameters can have type hints (I think the format is `<int:role_id>` and, given they're going to be treated as ints by the `role_admin_required` decorator we should probably use them